src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java

```diff
@@ -2373,6 +2373,16 @@ public Object getObject(int columnIndex) throws SQLServerException {
             returnValue = getTime(columnIndex);
         } else if (type == java.sql.Timestamp.class) {
             returnValue = getTimestamp(columnIndex);
+        } else if (type == java.time.LocalDateTime.class) {
+            returnValue = null;
+            java.sql.Timestamp ts = getTimestamp(columnIndex,
+                    Calendar.getInstance(java.util.TimeZone.getTimeZone("UTC")));
+            if (ts != null) {
+                java.time.format.DateTimeFormatter dtf = java.time.format.DateTimeFormatter
```

Comment:
Parsing to string and back is inefficient. Consider using `Timestamp.toLocalDateTime()` (assuming my previous comment is addressed), or `LocalDateTime.ofInstant(ts.getInstant(), <zone offset you want>)`.

Reply:
@mrotteveel Thanks. I went with the latter.